axelor-human-resource/src/main/java/com/axelor/apps/hr/db/repo/EmployeeHRRepository.java

```diff
@@ -169,4 +157,31 @@ public static boolean isEmployeeFormerNewOrArchived(Employee employee) {
                     .orElse(null));
     return isEmployeeFormerNewOrArchived(employee, today);
   }
+
+  protected String getEmployeeStatus(Employee employee) {
+    String employeeStatus = null;
+    LocalDate leavingDate = employee.getLeavingDate();
+    LocalDate hireDate = employee.getHireDate();
+
+    if (leavingDate == null && hireDate == null) {
+      return employeeStatus;
+    }
+
+    LocalDate today =
+        appBaseService.getTodayDate(
+            employee.getUser() != null
+                ? employee.getUser().getActiveCompany()
+                : AuthUtils.getUser().getActiveCompany());
+
+    if (employee.getLeavingDate() == null
+        && employee.getHireDate() != null
+        && employee.getHireDate().compareTo(today.minusDays(30)) > 0) {
+      employeeStatus = "new";
+    } else if (leavingDate != null && leavingDate.compareTo(today) < 0) {
+      employeeStatus = "former";
+    } else {
+      employeeStatus = "active";
+    }
+    return employeeStatus;
+  }
```

Comment:
Please create a new service EmployeeComputeStatusService for this method. It will be called with a Beans.get() in the repository.